common/lib/driver_dialect/driver_dialect.ts

```diff
@@ -27,4 +27,12 @@ export interface DriverDialect {
   preparePoolClientProperties(props: Map<string, any>, poolConfig: AwsPoolConfig | undefined): any;
 
   getAwsPoolClient(props: any): AwsPoolClient;
+
+  setConnectTimeout(props: Map<string, any>): void;
+
+  setupInitialNetworkTimeouts(props: Map<string, any>): void;
```

Comment:
What means "initial"? How it's different from queryTimeout?

Reply:
This method is to set up any properties used in the client configuration. The separate set connect/query timeout methods are from a [suggestion](https://github.com/aws/aws-advanced-nodejs-wrapper/pull/342#discussion_r1870194732), they may or may not be called in `setupInitialNetworkTimeouts`.

Reply:
I am leaning towards renaming this to setNetworkTimeouts since this method can technically be called any time during the application workflow.